libutils/known_dirs.c

```diff
@@ -98,13 +98,16 @@ const char *GetDefault##FUNC##Dir(void)                             \
     return GetDefaultDir_helper(STATIC##dir, GLOBAL, FOLDER);       \
 }
 
+#if !defined(__CYGWIN__) && !defined(__ANDROID__)
```

Comment:
Why `__CYGWIN__` and not `__MINGW32__`?

Reply:
Because line 31

```
#if defined(__CYGWIN__) || defined(__ANDROID__)
```

defines and uses GET_DEFAULT_DIRECTORY_DEFINE to define such functions as GetDefaultWorkDir().

But MINGW32 needs this block which uses the OTHER GET_DEFAULT_DIRECTORY_DEFINE() macro which uses the GetDefaultDir_helper() function, in MINGW32 case that's in enterprise. :|

So I would guess libntech won't build with `__MINGW32__` and maybe we should handle this better? Maybe move some or all of this file back to core to live along-side enterprise better?

Reply:
> So I would guess libntech won't build with `__MINGW32__` and maybe we should handle this better? Maybe move some or all of this file back to core to live along-side enterprise better?

Yeah, it is definitely not ideal, but that's out of scope of this PR.